test/integration/scheduler/scheduler_test.go

```diff
@@ -1314,6 +1316,62 @@ var _ = ginkgo.Describe("Scheduler", func() {
 		})
+
+		ginkgo.It("Pending workload with StrictFIFO doesn't block other CQ from borrowing from a third CQ", func() {
+			ginkgo.By("Creating ClusterQueues and LocalQueues")
+			cqTeamA := testing.MakeClusterQueue("team-a").
+				ResourceGroup(*testing.MakeFlavorQuotas(onDemandFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+				Cohort(chName).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqTeamA)).Should(gomega.Succeed())
+			defer util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cqTeamA, true)
+
+			strictFIFOLocalQueue := testing.MakeLocalQueue("strict-fifo-q", matchingNS.Name).ClusterQueue(strictFIFOClusterQ.Name).Obj()
+			lqTeamA := testing.MakeLocalQueue("team-a-lq", matchingNS.Name).ClusterQueue(cqTeamA.Name).Obj()
+
+			gomega.Expect(k8sClient.Create(ctx, strictFIFOLocalQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, lqTeamA)).Should(gomega.Succeed())
+
+			cqSharedResources := testing.MakeClusterQueue("shared-resources").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(onDemandFlavor.Name).Resource(corev1.ResourceCPU, "5").Obj()).
+				Cohort(chName).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqSharedResources)).Should(gomega.Succeed())
+			defer util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cqSharedResources, true)
+
+			ginkgo.By("Creating workloads")
+			admittedWl1 := testing.MakeWorkload("wl", matchingNS.Name).Queue(strictFIFOLocalQueue.
+				Name).Request(corev1.ResourceCPU, "3").Priority(10).Obj()
+			gomega.Expect(k8sClient.Create(ctx, admittedWl1)).Should(gomega.Succeed())
+
+			admittedWl2 := testing.MakeWorkload("player1-a", matchingNS.Name).Queue(lqTeamA.
+				Name).Request(corev1.ResourceCPU, "5").Priority(1).Obj()
+			gomega.Expect(k8sClient.Create(ctx, admittedWl2)).Should(gomega.Succeed())
+
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, admittedWl1, admittedWl2)
+			cqTeamA.Spec.StopPolicy = ptr.To(kueue.Hold)
+
+			pendingWl := testing.MakeWorkload("pending-wl", matchingNS.Name).Queue(strictFIFOLocalQueue.
+				Name).Request(corev1.ResourceCPU, "3").Priority(9).Obj()
+			gomega.Expect(k8sClient.Create(ctx, pendingWl)).Should(gomega.Succeed())
+
+			borrowingWl := testing.MakeWorkload("player2-a", matchingNS.Name).Queue(lqTeamA.
+				Name).Request(corev1.ResourceCPU, "5").Priority(11).Obj()
+			gomega.Expect(k8sClient.Create(ctx, borrowingWl)).Should(gomega.Succeed())
+
+			blockedWl := testing.MakeWorkload("player3-a", matchingNS.Name).Queue(lqTeamA.
+				Name).Request(corev1.ResourceCPU, "1").Priority(10).Obj()
+			gomega.Expect(k8sClient.Create(ctx, blockedWl)).Should(gomega.Succeed())
+
+			cqTeamA.Spec.StopPolicy = ptr.To(kueue.None)
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, pendingWl, blockedWl)
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, borrowingWl)
+
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, matchingNS)).To(gomega.Succeed())
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
```

Comment:
This is already done in `AfterEach`.

Reply:
This is needed to cleanup CQs

Reply:
ah ok, because you create the additional CQs.

However, these lines won't run if the test fails. You can put these in a `defer` function right after the CQs are created.

Reply:
yep, that what I had initially.